# Mini Project 1
## Compute the number of schools and, using pandas, min, max, standard deviation and mean of the amount of grants that the Philadelphia School District received in 2010-11



A data frame is a container that can hold data of different types.  It is a two dimensional data structure with data organized in a tabular fashion.

A panda series is a list; a single element of data created from manipulating a data frame which allows you to use panda methods on those results.  The user can tell a series from a Data Frame because when the results are displayed the last line is not data - it is information about the data. It is also only a one dimensional array.

This exercise will use pandas to import data into a data frame.  Then the data will be modified so it can be manipulated.  The results for the Philadelphia school district will be isolated and summarized.

Step 1:  Import the school funding data as a dataframe using panda then look at the first 6 rows of the data using the head function.

In [1]:
import pandas as pd
school_data = pd.read_csv('userssharedsdfschoolimprovement2010grants.csv')
school_data.head()

School Name      City State  \
0  HOGARTH KINGEEKUK MEMORIAL SCHOOL  SAVOONGA    AK   
1                    AKIACHAK SCHOOL  AKIACHAK    AK   
2                     GAMBELL SCHOOL   GAMBELL    AK   
3               BURCHELL HIGH SCHOOL   WASILLA    AK   
4                       AKIAK SCHOOL     AKIAK    AK   

                               District Name 2010/11/Award Amount  \
0              BERING STRAIT SCHOOL DISTRICT           $471014.00   
1                     YUPIIT SCHOOL DISTRICT           $520579.00   
2              BERING STRAIT SCHOOL DISTRICT           $449592.00   
3  MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT           $641184.00   
4                     YUPIIT SCHOOL DISTRICT           $399686.00   

   Model Selected                                           Location  
0  Transformation  200 MAIN ST\nSAVOONGA, AK 99769\n(63.6687, -17...  
1  Transformation  AKIACHAK 51100\nAKIACHAK, AK 99551\n(60.8911, ...  
2  Transformation  169 MAIN ST\nGAMBELL, AK 99742\n(63.7413, -171...  
3  Transformation  1775 WEST PARKS HWY\nWASILLA, AK 99654\n(61.57...  
4  Transformation     AKIAK 5227\nAKIAK, AK 99552\n(60.8879, -161.2)

Step 2:  Look at the types of data in the data frame.

In [2]:
school_data.dtypes

School Name             object
City                    object
State                   object
District Name           object
2010/11/Award Amount    object
Model Selected          object
Location                object
dtype: object

Step 3:  Rename the columns to eliminate spaces.

In [3]:
school_data.columns =school_data.columns.str.replace(' ', '_')
school_data.columns

Index(['School_Name', 'City', 'State', 'District_Name', '2010/11/Award_Amount',
       'Model_Selected', 'Location'],
      dtype='object')

Step 4:  Change the name of the Amount field.

In [5]:
school_data.rename(columns = {'2010/11/Award_Amount':'Amount'}, inplace=True)
school_data.columns

Index(['School_Name', 'City', 'State', 'District_Name', 'Amount',
       'Model_Selected', 'Location'],
      dtype='object')

Step 5:  Remove the $ in the rows in the Amount field so that it can be converted to a number.

In [6]:
school_data['Amount'] = school_data['Amount'].str.replace('$','')
school_data.head()

School_Name      City State  \
0  HOGARTH KINGEEKUK MEMORIAL SCHOOL  SAVOONGA    AK   
1                    AKIACHAK SCHOOL  AKIACHAK    AK   
2                     GAMBELL SCHOOL   GAMBELL    AK   
3               BURCHELL HIGH SCHOOL   WASILLA    AK   
4                       AKIAK SCHOOL     AKIAK    AK   

                               District_Name     Amount  Model_Selected  \
0              BERING STRAIT SCHOOL DISTRICT  471014.00  Transformation   
1                     YUPIIT SCHOOL DISTRICT  520579.00  Transformation   
2              BERING STRAIT SCHOOL DISTRICT  449592.00  Transformation   
3  MATANUSKA-SUSITNA BOROUGH SCHOOL DISTRICT  641184.00  Transformation   
4                     YUPIIT SCHOOL DISTRICT  399686.00  Transformation   

                                            Location  
0  200 MAIN ST\nSAVOONGA, AK 99769\n(63.6687, -17...  
1  AKIACHAK 51100\nAKIACHAK, AK 99551\n(60.8911, ...  
2  169 MAIN ST\nGAMBELL, AK 99742\n(63.7413, -171...  
3  1775 WEST PARKS HWY\nWASILLA, AK 99654\n(61.57...  
4     AKIAK 5227\nAKIAK, AK 99552\n(60.8879, -161.2)

Step 6:  Change the Amount field to a number field.

In [7]:
school_data['Amount'] = school_data.Amount.astype(float)
school_data.dtypes

School_Name        object
City               object
State              object
District_Name      object
Amount            float64
Model_Selected     object
Location           object
dtype: object

Step 7: Filter on just the records for the Philadelphia City SD

In [8]:
philly = school_data["District_Name"].str.contains("PHILADELPHIA CITY SD")
school_data[philly]

School_Name          City State  \
603               SAYRE WILLIAM L MS  PHILADELPHIA    PA   
605            KING MARTIN LUTHER HS  PHILADELPHIA    PA   
606            DOUGLAS STEPHEN A SCH  PHILADELPHIA    PA   
609                  LOCKE ALAIN SCH  PHILADELPHIA    PA   
610                 ALCORN JAMES SCH  PHILADELPHIA    PA   
613                    GERMANTOWN HS  PHILADELPHIA    PA   
616                    ROXBOROUGH HS  PHILADELPHIA    PA   
620               UNIVERSITY CITY HS  PHILADELPHIA    PA   
621             SMEDLEY FRANKLIN SCH  PHILADELPHIA    PA   
624             WEST PHILADELPHIA HS  PHILADELPHIA    PA   
625            HARRITY WILLIAM F SCH  PHILADELPHIA    PA   
628                     FRANKFORD HS  PHILADELPHIA    PA   
631                  VAUX ROBERTS HS  PHILADELPHIA    PA   
633              CLEMENTE ROBERTO MS  PHILADELPHIA    PA   
634                   FELS SAMUEL HS  PHILADELPHIA    PA   
636               ALLEN DR ETHEL SCH  PHILADELPHIA    PA   
637  FELTONVILLE INTERMEDIATE SCHOOL  PHILADELPHIA    PA   
640            SOUTH PHILADELPHIA HS  PHILADELPHIA    PA   
641                    OLNEY HS WEST  PHILADELPHIA    PA   
643         EDISON HS/FAREIRA SKILLS  PHILADELPHIA    PA   
646                STETSON JOHN B MS  PHILADELPHIA    PA   
648                DAROFF SAMUEL SCH  PHILADELPHIA    PA   
649             BLUFORD GUION EL SCH  PHILADELPHIA    PA   
651                CLYMER GEORGE SCH  PHILADELPHIA    PA   
652               MANN WILLIAM B SCH  PHILADELPHIA    PA   
655                   GRATZ SIMON HS  PHILADELPHIA    PA   
658                       LINCOLN HS  PHILADELPHIA    PA   

            District_Name    Amount  Model_Selected  \
603  PHILADELPHIA CITY SD  737032.0  Transformation   
605  PHILADELPHIA CITY SD  786964.0  Transformation   
606  PHILADELPHIA CITY SD  319320.0         Restart   
609  PHILADELPHIA CITY SD  390938.0  Transformation   
610  PHILADELPHIA CITY SD  338726.0  Transformation   
613  PHILADELPHIA CITY SD  798792.0  Transformation   
616  PHILADELPHIA CITY SD  641440.0  Transformation   
620  PHILADELPHIA CITY SD  923999.0      Turnaround   
621  PHILADELPHIA CITY SD  362773.0         Restart   
624  PHILADELPHIA CITY SD  671142.0  Transformation   
625  PHILADELPHIA CITY SD  980283.0         Restart   
628  PHILADELPHIA CITY SD  792001.0  Transformation   
631  PHILADELPHIA CITY SD  466665.0      Turnaround   
633  PHILADELPHIA CITY SD  333327.0      Turnaround   
634  PHILADELPHIA CITY SD  974132.0  Transformation   
636  PHILADELPHIA CITY SD  247020.0      Turnaround   
637  PHILADELPHIA CITY SD  244855.0  Transformation   
640  PHILADELPHIA CITY SD  579958.0  Transformation   
641  PHILADELPHIA CITY SD  272245.0  Transformation   
643  PHILADELPHIA CITY SD       1.0  Transformation   
646  PHILADELPHIA CITY SD  888708.0         Restart   
648  PHILADELPHIA CITY SD  658247.0         Restart   
649  PHILADELPHIA CITY SD  907129.0         Restart   
651  PHILADELPHIA CITY SD  392068.0  Transformation   
652  PHILADELPHIA CITY SD  794021.0         Restart   
655  PHILADELPHIA CITY SD  508486.0  Transformation   
658  PHILADELPHIA CITY SD      14.0  Transformation   

                                              Location  
603  5800 WALNUT ST\nPHILADELPHIA, PA 19139\n(39.95...  
605  6100 STENTON AVE\nPHILADELPHIA, PA 19138\n(40....  
606  2700 EAST HUNTINGDON ST\nPHILADELPHIA, PA 1912...  
609  4550 HAVERFORD AVE\nPHILADELPHIA, PA 19139\n(3...  
610  3200 DICKINSON ST\nPHILADELPHIA, PA 19146\n(39...  
613  40 EAST HIGH ST\nPHILADELPHIA, PA 19144\n(40.0...  
616  6498 RIDGE AVE\nPHILADELPHIA, PA 19128\n(40.03...  
620  3601 FILBEROUTE ST\nPHILADELPHIA, PA 19104\n(3...  
621  1790 BRIDGE ST\nPHILADELPHIA, PA 19124\n(40.01...  
624  4700 WALNUT ST\nPHILADELPHIA, PA 19139\n(39.95...  
625  5601 CHRISTIAN ST\nPHILADELPHIA, PA 19143\n(39...  
628  5000 OXFORD AVE\nPHILADELPHIA, PA 19124\n(40.0...  
631  2300 WEST MASTER ST\nPHILADELPHIA, PA 19121\n(...  
633  

Step 8:  Calculate the number of schools in the Philadelphia City SD and min, max, mean and standard deviation of the funding.

In [9]:
school_data[philly].describe()

Amount
count      27.000000
mean   555936.518519
std    288151.681054
min         1.000000
25%    336026.500000
50%    579958.000000
75%    793011.000000
max    980283.000000

There are 27 schools that received funding.  The minimum amount was 1.  The maximum amount was 980,283.  The mean amount was 555,936.52.  The standard deviation was 288,151.68.  Since I used only pandas functions, the standard deviation that pandas calculates is sample standard deviation.